api/src/opentrons/protocol_api/core/engine/deck_conflict.py

```diff
@@ -174,6 +183,57 @@ def check_safe_for_pipette_movement(
         )
 
 
+def check_safe_for_tip_pickup_and_return(
+    engine_state: StateView,
+    pipette_id: str,
+    labware_id: str,
+) -> None:
+    """Check if the presence or absence of a tiprack adapter might cause any pipette movement issues.
+
+    A 96 channel pipette will pick up tips using cam action when it's configured
+    to use ALL nozzles. For this, the tiprack needs to be on the Flex 96 channel tiprack adapter
+    or similar or the tips will not be picked up.
+
+    On the other hand, if the pipette is configured with partial nozzle configuration,
+    it uses the usual pipette presses to pick the tips up, in which case, having the tiprack
+    on the Flex 96 channel tiprack adapter (or similar) will cause the pipette to
+    crash against the adapter posts.
+
+    In order to check if the 96-channel can move and pickup/drop tips safely, this method
+    checks for the height attribute of the tiprack adapter rather than checking for the
+    specific official adapter since users might create custom labware &/or definitions
+    compatible with the official adapter.
+    """
+    if not engine_state.pipettes.get_channels(pipette_id) == 96:
+        # Adapters only matter to 96 ch.
+        return
+
+    is_partial_config = engine_state.pipettes.get_is_partially_configured(pipette_id)
+    tiprack_name = engine_state.labware.get_load_name(labware_id)
+    tiprack_parent = engine_state.labware.get_location(labware_id)
+    if isinstance(tiprack_parent, OnLabwareLocation):  # tiprack is on an adapter
+        tiprack_height = engine_state.labware.get_dimensions(labware_id).z
+        adapter_height = engine_state.labware.get_dimensions(tiprack_parent.labwareId).z
+        if is_partial_config and tiprack_height < adapter_height:
+            raise PartialTipMovementNotAllowedError(
+                f"Tiprack {tiprack_name} cannot be on an adapter taller than the tiprack"
+                f"  when the 96-channel is configured for using partial nozzles."
+            )
+        elif not is_partial_config and tiprack_height > adapter_height:
+            raise PipetteMovementNotAllowed(
+                f"Tiprack {tiprack_name} must be on a Flex Tiprack Adapter (or similar)"
+                f" in order to pick up or return tips with all 96-channel nozzles."
```

Comment:
```suggestion
                f"Tip rack {tiprack_name} must be on an Opentrons Flex 96 Tip Rack Adapter"
                f" in order to pick up or return 96 tips simultaneously."
```
Also paralleling language used elsewhere. I think we should refer to the specific labware, or we could be promising more than the hardware API can handle. For example, you have some custom adapter that's just like ours but 5 mm taller…do you crash? is the force correct? etc. It may not Just Work™.

And a fact check: can you `return_tip()` when config is set as `ALL`? Is it only disabled for partial config?

Reply:
Only disabled for partial config